sklearn/impute/_base.py

```diff
@@ -366,7 +375,7 @@ def transform(self, X):
                              % (X.shape[1], self.statistics_.shape[0]))
 
         if self.add_indicator:
-            X_trans_indicator = self.indicator_.transform(X)
+            X_trans_indicator = self.indicator_.transform(self._missing_mask)
```

Comment:
This it the mask of the data at `fit` and not here at `transform`. We should add a test to make sure the following gives the expected result

```
imp = SimpleImputer(..., add_indicator=True)
imp.fit(X_train)
imp.transform(X_test)
```

Reply:
oh yeah, I totally ignored it, sure will do.

Thanks.